msbuild/Xamarin.MacDev.Tasks/Tasks/ComputeBundleLocationTaskBase.cs

```diff
@@ -119,16 +120,23 @@ public override bool Execute ()
 					continue;
 				case PublishFolderType.CompressedAppleFramework:
 					relativePath = FrameworksDirectory;
+					virtualProjectPath = Path.GetFileNameWithoutExtension (item.ItemSpec);
+					if (virtualProjectPath.EndsWith (".xcframework", StringComparison.OrdinalIgnoreCase))
+						virtualProjectPath = Path.ChangeExtension (virtualProjectPath, ".framework");
 					break;
 				case PublishFolderType.AppleBindingResourcePackage:
+					// Nothing to do here, this is handled fully in the targets file
+					break;
 				case PublishFolderType.CompressedAppleBindingResourcePackage:
 					// Nothing to do here, this is handled fully in the targets file
```

Comment:
Question - why use a hand-rolled `RemoveExtension` when there exists `Path.GetFileNameWithoutExtention ()`? Is it because you only want to remove the extension if it ends in `.zip`?

Reply:
Because Path.GetFileNameWithoutExtension would remove any path parts as well, which we want to keep.

We want: "some/path/foo.zip" => "some/path/foo", while Path.GetFileNameWithoutExtension would return just "foo".